lambdas/catalog-updater/index.js

```diff
@@ -153,31 +170,46 @@ exports.handler = (event, context) => {
     let params = {
       Bucket: bucketName
     },
-    promises = []
+      promises = []
 
     return s3.listObjectsV2(params).promise()
       .then((result) => {
         console.log(`result: ${JSON.stringify(result, null, 4)}`)
         let promises =
           _
-          .chain(result.Contents)
-          .filter(swaggerFileFilter)
-          .map(getSwaggerFile)
-          .value()
+            .chain(result.Contents)
+            .filter(swaggerFileFilter)
+            .map(getSwaggerFile)
+            .value()
 
         return Promise.all(promises)
       })
       .then((swaggerFiles) => {
         console.log(`results: ${JSON.stringify(swaggerFiles, null, 4)}`)
-        let catalogObjects = []
+        let catalogObjects = {
+          apiGateway: [],
+          generic: []
+        }
 
         return gateway.getUsagePlans({}).promise()
           .then((result) => {
             console.log(`usagePlans: ${JSON.stringify(result.items, null, 4)}`)
             let usagePlans = result.items
             for (let i = 0; i < usagePlans.length; i++) {
-              catalogObjects[i] = usagePlanToCatalogObject(usagePlans[i], swaggerFiles)
+              catalogObjects.apiGateway[i] = usagePlanToCatalogObject(
+                usagePlans[i], swaggerFiles.filter(s => !s.generic))
             }
+
+            catalogObjects.generic.push(
+              ...swaggerFiles.filter(s => s.generic).map(s => {
+                //TODO: Allow for customizing image?
+                s.image = '/sam-logo.png'
```

Comment:
Not used.

```suggestion
```

Reply:
It's necessary or client side throws - if we intend on letting users modify images associated with APIs in the future, I don't see the harm in leaving this in place.

Reply:
It won't now (b/c you merged in the changes in from staging.)